pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -405,11 +407,17 @@ def training_type_plugin(self) -> TrainingTypePlugin:
         # attach checkpoint plugin to the training type plugin
         if self._checkpoint_io is not None:
             self._training_type_plugin.checkpoint_io = self._checkpoint_io
-        precision_plugin = self.precision_plugin
-        if precision_plugin is not None:
-            self._training_type_plugin._precision_plugin = precision_plugin
+        if (hasattr(self.strategy, "precision_plugin") and self.precision_plugin is None) or not hasattr(
+            self.strategy, "precision_plugin"
+        ):
+            precision_plugin = self.precision_plugin
+            if precision_plugin is not None:
+                self._training_type_plugin._precision_plugin = precision_plugin
         self._training_type_plugin_resolved = True
-
+        if (hasattr(self.strategy, "accelerator") and self.strategy.accelerator is None) or not hasattr(
+            self.strategy, "accelerator"
+        ):
+            self._training_type_plugin.accelerator = self.select_accelerator()
```

Comment:
@awaelchli this is really ugly, but this is the minimal impact way I can think about to handle custom plugins. How about we refactor/rewrite Accelerator_connector after this and keep logic as same as possible compare to before for now?